### PR TITLE
Define new well-known annotation: kubernetes.io/display-name

### PR DESCRIPTION
**Pull request by akutz**
This patch introduces a new, well-known annotation, `kubernetes.io/display-name`. Please refer to https://github.com/kubernetes/kubernetes/issues/121514 for the background, goals, and non-goals. Thanks!

### CODE REVIEW

**Comment by akutz** [comment]
/assign reylejano

cc @sftim

**Comment by sftim** [comment]
/sig architecture
/unassign

I'm happy for the first available reviewer to work on this.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (aj11anuj) [inline content/en/docs/reference/labels-annotations-taints/_index.md:546]:
```suggestion
Example: `kubernetes.io/display-name: "Displays the name of a K8s object."`
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (aj11anuj) [inline content/en/docs/reference/labels-annotations-taints/_index.md:552]:
```suggestion
the value for this annotation is mutable and does not need to comply with a DNS
```

**Review by aj11anuj — CHANGES_REQUESTED** [review]
Some minor suggestions from my side

**Comment by reylejano** [comment]
/hold
Putting a hold on this PR because an entry for a well-known annotation should not exist in kubernetes.io until the annotation exists
Once the annotation exists, a reviewer can test to confirm the annotation exists and SIG Arch can confirm the description is technically correct

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (mamachanko) [reply content/en/docs/reference/labels-annotations-taints/_index.md:546]: To me `"Display name of K8s object."` is the more accurate example. After all, it is the _display name_.

**Comment by sftim** [comment]
OK to unhold this once SIG Architecture have provided an LGTM tech review.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (akutz) [reply content/en/docs/reference/labels-annotations-taints/_index.md:546]: Sure, I was just copying the above example. This feels like a preference, but I'll go with it.

**Comment by akutz** [comment]
Closing per https://github.com/kubernetes/kubernetes/issues/121514#issuecomment-1813012286

**Pull request closed without merging**
